### PR TITLE
Feature/register client

### PR DESCRIPTION
**Pull request by dankelleher**

### CODE REVIEW

File: client/test/e2e/resolve.test.ts [hunk 1]
```diff
@@ -1,6 +1,6 @@
 import { resolve } from '../../src';
 import { SolanaUtil } from '../../src/solana-util';
-import { ClusterType } from '../../src/solid-data';
+import { getClusterType } from '../../src/solid-data';
```
Comment (joncinque) [inline client/test/e2e/resolve.test.ts:3]: I don't see where this function exists in the code -- should this be `ClusterType`?

File: client/test/e2e/resolve.test.ts [hunk 2]
```diff
@@ -24,7 +24,7 @@ describe('resolve', () => {
       connection,
       payer,
       authority.publicKey,
-      ClusterType.development()
+      getClusterType(CLUSTER)
```
Comment (joncinque) [inline client/test/e2e/resolve.test.ts:27]: And this just `CLUSTER`?

File: client/src/util.ts [hunk 3]
```diff
@@ -2,29 +2,66 @@ import { DIDDocument, PublicKey } from 'did-resolver';
 import { Account, PublicKey as SolanaPublicKey } from '@solana/web3.js';
 import { DID_METHOD } from './constants';
 import { ClusterType } from './solid-data';
+import { decode, encode } from 'bs58';
 
 // a 64-byte private key on the X25519 curve.
 // In string form it is base58-encoded
-export type PrivateKey = number[] | string | Buffer;
+export type PrivateKey = number[] | string | Buffer | Uint8Array;
```
Comment (joncinque) [inline client/src/util.ts:9]: I've been wondering about this type -- is it meant for when people want to sign Solana transactions using their private key, no matter what format it's in?

**Review by joncinque — COMMENTED** [review]
Looks great, just a couple of questions

File: driver/src/service/Registrar.ts [hunk 4]
```diff
@@ -1,9 +1,45 @@
...
+  if (!payer)
+    throw new Error('Missing payer information- add a request secret');
+
+  let ownerPublicKey = request.options?.owner;
+  let ownerPrivateKey = undefined;
+  if (!ownerPublicKey) {
+    ({
+      publicKey: ownerPublicKey,
+      secretKey: ownerPrivateKey,
+    } = DID.generateKeypair());
```
[the remark this answers is not in the document]
Comment (dankelleher) [reply driver/src/service/Registrar.ts:18]: Thats right, but we are returning the private key in the response, which is not particularly safe, but it is how the uniregistrar appears to work. I would think most people would instead choose to pass an owner in, but I wanted to remain compliant to the uniregistrar.

File: client/test/e2e/resolve.test.ts (hunk 1 above)
Comment (dankelleher) [reply client/test/e2e/resolve.test.ts]: Fixed

File: client/test/e2e/resolve.test.ts (hunk 2 above)
Comment (dankelleher) [reply client/test/e2e/resolve.test.ts]: Fixed

File: client/src/util.ts (hunk 3 above)
Comment (dankelleher) [reply client/src/util.ts:9]: Yes exactly. e.g. it allows you to pass in a base58 encoded string, which is easier for curl calls or command-line tools.

**Review by joncinque — APPROVED** [review]

**Pull request merged**